validator/api/src/main/java/tech/pegasys/teku/validator/api/ValidatorApiChannel.java

```diff
@@ -42,6 +42,133 @@
 import tech.pegasys.teku.spec.datastructures.validator.SubnetSubscription;
 
 public interface ValidatorApiChannel extends ChannelInterface {
+  ValidatorApiChannel NO_OP =
```

Comment:
I think returning `null` in the NOOP methods implementations breaks the `ValidatorApiChannel` interface. This can lead to errors and frustration in the future. My suggestion would be to respect the async and optional expectations in the interface. Something like this:

```
      new ValidatorApiChannel() {
        @Override
        public SafeFuture<Optional<GenesisData>> getGenesisData() {
          return SafeFuture.completedFuture(Optional.empty());
        }

        @Override
        public SafeFuture<Map<BLSPublicKey, Integer>> getValidatorIndices(final Collection<BLSPublicKey> publicKeys) {
          return SafeFuture.completedFuture(Map.of());
        }

        [...]
```